libraries/AP_HAL_PX4/RCOutput.cpp

```diff
@@ -399,34 +395,12 @@ void PX4RCOutput::read_last_sent(uint16_t* period_us, uint8_t len)
     }
 }
 
-/*
-  update actuator armed state
- */
-void PX4RCOutput::_arm_actuators(bool arm)
-{
-    if (_armed.armed == arm) {
-        // already armed;
-        return;
-    }
-
-	_armed.timestamp = hrt_absolute_time();
-    _armed.armed = arm;
-    if (arm) {
-        // this latches ready_to_arm to true once set once. Otherwise
-        // we have a race condition with px4io safety switch update
-        _armed.ready_to_arm = true;
-    }
-    _armed.lockdown = false;
-    _armed.force_failsafe = false;
-}
-
 void PX4RCOutput::_send_outputs(void)
 {
     uint32_t now = AP_HAL::micros();
 
     if ((_enabled_channels & ((1U<<_servo_count)-1)) == 0) {
         // no channels enabled
-        _arm_actuators(false);
```

Comment:
I'm not sure why this was disabling the UAVCAN actuators but not others. I'm not even sure this _if_ is correct - I never understood what this enable/disable channel is supposed to do; if I didn't miss anything once we enable a channel we don't ever disable it (I didn't found any calls to *disable_ch*).
@tridge?

@EShamaev I'm glad you pointed me at that line though. I think there's a bug there, it should arm depending on safety switch state and not soft armed state. I've pushed a commit here to fix that.